### PR TITLE
Added TOC and tidy-up readme

### PR DESCRIPTION
**Pull request by alvinometric**
This is to make sure others don't miss the Rails instructions, like I did (and apparently many of customers still do)

@melindafekete I added `1.a`, `1.b`, etc to show that there are different options, but happy to hear another idea

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -315,20 +350,6 @@ if Rails.configuration.unleash.is_enabled? "AwesomeFeature", @unleash_context
 end
 ```
 
-If you don't want to check a feature is disabled with `unless`, you can also use `is_disabled?`:
```
Comment (gardleopard) [inline README.md:318]: Why is this example removed?

File: README.md [hunk 2]
```diff
@@ -134,29 +159,34 @@ end
 
 ## Usage in a Rails Application
 
-#### Add Initializer
+### 1.a Add Initializer
 
 Put in `config/initializers/unleash.rb`:
 
 ```ruby
+require 'unleash'
```
Comment (gardleopard) [inline README.md:167]: why is this added?

File: README.md [hunk 3]
```diff
@@ -134,29 +159,34 @@ end
...
+
 Unleash.configure do |config|
   config.app_name = Rails.application.class.parent.to_s
   config.url      = 'https://unleash.herokuapp.com/api'
   # config.instance_id = "#{Socket.gethostname}"
   config.logger   = Rails.logger
+  config.custom_http_headers = {'Authorization': '<YOUR_API_TOKEN>'}
```
Comment (gardleopard) [inline README.md:174]: Nice!

File: README.md [hunk 4]
```diff
@@ -297,9 +332,9 @@ Add in `app/controllers/application_controller.rb`:
 
 Or if you see better fit, only in the controllers that you will be using unleash.
 
-#### Sample usage
+### 3. Sample usage
 
-Then wherever in your application that you need a feature toggle, you can use:
```
Comment (rarruda) [inline README.md:302]: There is no requirement to use rails in order to use unleash. The instructions are valid no matter which (if any) framework is used.

File: README.md [hunk 5]
```diff
@@ -237,7 +272,7 @@ end
 
 Note that we also added shutdown hooks in `on_worker_shutdown`, to ensure a clean shutdown.
 
-#### Add Initializer if using [Phusion Passenger](https://github.com/phusion/passenger)
+#### 1.c Add Initializer if using [Phusion Passenger](https://github.com/phusion/passenger)
```
Comment (melindafekete) [inline README.md:275]:
```suggestion
### 1.c Add Initializer if using [Phusion Passenger](https://github.com/phusion/passenger)
```

File: README.md [hunk 6]
```diff
@@ -259,7 +294,7 @@ PhusionPassenger.on_event(:starting_worker_process) do |forked|
 end
 ```
 
-#### Add Initializer hooks when using within [Sidekiq](https://github.com/mperham/sidekiq)
+### Add Initializer hooks when using within [Sidekiq](https://github.com/mperham/sidekiq)
```
Comment (melindafekete) [inline README.md:297]:
```suggestion
### 1.d Add Initializer hooks when using within [Sidekiq](https://github.com/mperham/sidekiq)
```

**Review by melindafekete — COMMENTED** [review]
@alvinometric I think the number/letter list for Initializer works, but perhaps we could add a heading level above the steps, and add a sentence to explain. Something like

```
### 1. Add Initializer

The initializer setup varies depending on whether you’re using a standard setup, Puma in clustered mode, Phusion Passenger, or Sidekiq.
 
#### 1.a Initializer for standard Rails applications
#### 1.b Initializer for Puma in clustered mode
#### 1.c Initializer for Phusion Passenger
#### 1.d Initializer for Sidekiq
```

File: README.md (hunk 1 above)
Comment (sighphyre) [reply README.md:318]: I'm with Gard, I'd like this to stay, it's part of the public API and needs to be surfaced here

File: README.md [hunk 7]
```diff
@@ -297,9 +332,9 @@ Add in `app/controllers/application_controller.rb`:
 
 Or if you see better fit, only in the controllers that you will be using unleash.
 
-#### Sample usage
+### 3. Sample usage
```
Comment (sighphyre) [inline README.md:335]: I don't think this is section 3, this is general usage for the client

Comment (alvinometric) [reply README.md:335]: Right now it's a subsection of the Rails section, that's why it's an h3, but we can make another h2 for general usage

**Comment by alvinometric** [comment]
> @alvinometric I think the number/letter list for Initializer works, but perhaps we could add a heading level above the steps, and add a sentence to explain. Something like
> 
> ```
> ### 1. Add Initializer
> 
> The initializer setup varies depending on whether you’re using a standard setup, Puma in clustered mode, Phusion Passenger, or Sidekiq.
>  
> #### 1.a Initializer for standard Rails applications
> #### 1.b Initializer for Puma in clustered mode
> #### 1.c Initializer for Phusion Passenger
> #### 1.d Initializer for Sidekiq
> ```

Good call, added 🙏

File: README.md (hunk 1 above)
Comment (alvinometric) [reply README.md:318]: There was no other mention of using `unless` anywhere in the document, so it felt weird to say "if you don't want to use this thing I never told you about"

Comment (gardleopard) [reply README.md:318]: unless is built in to ruby, the example is for `is_disabled?` usage

**Review by gardleopard — APPROVED** [review]
LGTM, good job!
